module-4/hashcat/README.md

```diff
@@ -0,0 +1,95 @@
+# My `hashcat` tool
+
+This folder contains the source code for my own implementation of the `hashcat` tool.
+Hashcat can be used to TBD.
+
+This is a lightweight implementation that covers the following features/options:
+
+- `-m` `--mode`, Hash mode: 0 (MD5), 1 (SHA-1), 2 (SHA-256), 3 (SHA-512)
+- `-a` `--attack`, Attack mode: 0 (Brute-Force), 1 (Dictionary)
+- `-h` `--hash`, Target hash (required if -H is not used)
+- `-H` `--hashfile`, File containing the target hash (required if -h is not used)
+- `-w` `--wordlist`,Path to the wordlist file (required for Dictionary attack)
+- `--help`, Show this help message and exit
```

Comment:
as for hydra, better use a table element and add a column that indicates if the option is mandatory or not.